unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/VoiceChatHUD/VoiceChatWindowController.cs

```diff
@@ -161,40 +164,41 @@ public void Dispose()
 
     internal void CloseView() { SetVisibility(false); }
 
-    internal void JoinVoiceChat(bool isJoined)
-    { 
-        voiceChatWindowView.SetAsJoined(isJoined);
-
+    internal void RequestJoinVoiceChat(bool isJoined)
+    {
         if (isJoined)
         {
-            voiceChatBarView.Show();
-            SetWhichPlayerIsTalking();
+            WebInterface.JoinVoiceChat();
         }
         else
         {
-            dataStore.voiceChat.isRecording.Set(new KeyValuePair<bool, bool>(false, false), true);
-            isOwnPLayerTalking = false;
-            voiceChatBarView.Hide();
+            if (dataStore.voiceChat.isRecording.Get().Key)
```

Comment:
you can extract `dataStore.voiceChat.isRecording` into a variable so its more clean